server/container/util-java8/src/main/java/org/apache/james/util/FunctionGenerator.java

```diff
@@ -0,0 +1,29 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.util;
+
+import java.util.function.Function;
+
+@FunctionalInterface
+public interface FunctionGenerator<A, R> extends Function<A, Function<R, R>> {
```

Comment:
Wait, the class basically add a method that returns a function : 
```java
this.apply(a); //Do nothing with the result
return other.apply(a);
```
That's it ?

Then `this` is not a `Function` but a `Consumer`, right ?